app/src/main/java/org/wikipedia/util/StringUtil.kt

```diff
@@ -105,6 +106,16 @@ object StringUtil {
                 == Normalizer.normalize(str2, Normalizer.Form.NFC))
     }
 
+    fun extractHrefLinks(source: String?): MutableList<String> {
```

Comment:
We don't need to create our own parser for href links. We already use a powerful library called `Jsoup` elsewhere, that you can use to extract link elements with one line of code.